cpp/ql/src/experimental/Best Practices/UselessTest.ql

```diff
@@ -0,0 +1,50 @@
+/**
+ * @name Useless Test
+ * @description Find any useless test of kind a==8 && a!=7
+ * @kind problem
+ * @problem.severity warning
+ * @id cpp/uselesstest
+ * @tags reliability
+ *       readability
+ */
+
+import cpp
+
+predicate sameStmt(Expr e1, Expr e2) {
```

Comment:
I'm not quite sure I understand the meaning of this predicate. The name makes it sound like you want to check if two statements are equivalent, but the two parameters are of type `Expr` (and not `Stmt`)?

If what you want is to compare two expressions, I'd consider trying out our global value numbering library (by importing `semmle.code.cpp.valuenumbering.GlobalValueNumbering`). Then you can check if two expressions are guarenteed to evaluate to the same value by check if `globalValueNumber(e1).getAnExpr() = e2` holds. We also have a hash cons library which is a more syntactic version of global value numbering. You can read up on both libraries here: https://codeql.github.com/docs/codeql-language-guides/hash-consing-and-value-numbering/